indexing-service/src/main/java/io/druid/indexing/common/task/RestoreTask.java

```diff
@@ -62,52 +62,61 @@ public String getType()
   }
 
   @Override
-    public TaskStatus run(TaskToolbox toolbox) throws Exception
-    {
-      // Confirm we have a lock (will throw if there isn't exactly one element)
-      final TaskLock myLock = Iterables.getOnlyElement(getTaskLocks(toolbox));
+  public TaskStatus run(TaskToolbox toolbox) throws Exception
+  {
+    // Confirm we have a lock (will throw if there isn't exactly one element)
+    final TaskLock myLock = Iterables.getOnlyElement(getTaskLocks(toolbox));
 
-      if (!myLock.getDataSource().equals(getDataSource())) {
-        throw new ISE("WTF?! Lock dataSource[%s] != task dataSource[%s]", myLock.getDataSource(), getDataSource());
-      }
+    if (!myLock.getDataSource().equals(getDataSource())) {
+      throw new ISE("WTF?! Lock dataSource[%s] != task dataSource[%s]", myLock.getDataSource(), getDataSource());
+    }
 
-      if (!myLock.getInterval().equals(getInterval())) {
-        throw new ISE("WTF?! Lock interval[%s] != task interval[%s]", myLock.getInterval(), getInterval());
-      }
+    if (!myLock.getInterval().equals(getInterval())) {
+      throw new ISE("WTF?! Lock interval[%s] != task interval[%s]", myLock.getInterval(), getInterval());
+    }
 
-      // List unused segments
-      final List<DataSegment> unusedSegments = toolbox
-          .getTaskActionClient()
-          .submit(new SegmentListUnusedAction(myLock.getDataSource(), myLock.getInterval()));
-
-      // Verify none of these segments have versions > lock version
-      for (final DataSegment unusedSegment : unusedSegments) {
-        if (unusedSegment.getVersion().compareTo(myLock.getVersion()) > 0) {
-          throw new ISE(
-              "WTF?! Unused segment[%s] has version[%s] > task version[%s]",
-              unusedSegment.getIdentifier(),
-              unusedSegment.getVersion(),
-              myLock.getVersion()
-          );
-        }
-
-        log.info("OK to restore segment: %s", unusedSegment.getIdentifier());
+    // List unused segments
+    final List<DataSegment> unusedSegments = toolbox
+        .getTaskActionClient()
+        .submit(new SegmentListUnusedAction(myLock.getDataSource(), myLock.getInterval()));
+
+    // Verify none of these segments have versions > lock version
+    for (final DataSegment unusedSegment : unusedSegments) {
+      if (unusedSegment.getVersion().compareTo(myLock.getVersion()) > 0) {
+        throw new ISE(
+            "WTF?! Unused segment[%s] has version[%s] > task version[%s]",
+            unusedSegment.getIdentifier(),
+            unusedSegment.getVersion(),
+            myLock.getVersion()
+        );
       }
 
-      List<DataSegment> restoredSegments = Lists.newLinkedList();
+      log.info("OK to restore segment: %s", unusedSegment.getIdentifier());
+    }
 
-      // Move segments
-      for (DataSegment segment : unusedSegments) {
+    List<DataSegment> restoredSegments = Lists.newLinkedList();
```

Comment:
I think `LinkedList` is pointless here. It's not a performance-critical place; but there should always be a reason to use `LinkedList` rather than default `ArrayList`. When reader sees `LinkedList`, he spends mental cycles trying to understand, why it is used here. It's confusing when there is no apparent reason.